src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -984,6 +998,16 @@ private function addMessengerSection(ArrayNodeDefinition $rootNode)
                                 ->end()
                             ->end()
                         ->end()
+                        ->arrayNode('middlewares')
+                            ->addDefaultsIfNotSet()
+                            ->children()
+                            ->arrayNode('doctrine_transaction')
```

Comment:
This should not be configured via Framework, but via DoctrineBundle. We don't have any other Doctrine config here AFAIR.

Reply:
@fabpot are you happy about the middleware being the bridge, right?

Reply:
yes

Reply:
@fabpot actually, [the cache](https://github.com/Nyholm/symfony/blob/68131f43bc806c93cca6617d78d6f311539cc5d6/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L1149-L1153) also relies on Doctrine in `FrameworkBundle`. Note that this doesn't configure Doctrine either it justs allows enables the middleware. I'll submit a PR to remove it from here but I believe it might makes sense (follows the same pattern as cache configuration) to be here 🤔

Reply:
https://github.com/symfony/symfony/pull/26684 is in.

Reply:
I also think this makes sense as is. Moving it to doctrine bundle would mean configuring messenger middlewares in two different places, I can't think of a good final result.
We enable logging features for some components via framework config in the same way

Reply:
I agree with @chalasr , it's only a feature flag. It does no harm having this in the fwb whereas having to configure middlewares in 2 different config keys is really bad, DX wise. WDYT?